include/arch/arm/aarch64/scripts/linker.ld

```diff
@@ -114,39 +106,27 @@ SECTIONS
 
     _image_rom_start = ROM_ADDR;
 
-    SECTION_PROLOGUE(rom_start,,)
+    SECTION_PROLOGUE(_TEXT_SECTION_NAME,,)
```

Comment:
Is it possible for this to look like Cortex-M and Cortex-R? 
rom_start can include vector tables (At least) and other applicable section, and text can include all other stuff that belong to text

Reply:
> rom_start can include vector tables (At least) 

Not sure about all other sections for similarity but  vector table in particular for cortex-A / Cortex-R belong to .text  unlike Cortex-M. So it differs there.

Reply:
For Cortex-R, vector table section is in rom_start, please, check.
Why can't you do the same for Cortex-A?

Reply:
For ARMv8-A, vector table is added in `.exc_vector_table` section, and on interrupt/exception core executes from there, so we have to put it in text segment.

Reply:
Agreed; can you just do the same thing as Cortex-M and Cortex-R? vector section is inside _exc_vector_table

Reply:
@Abhishek-brcm see the vector_table.ld in arm/aarrch32

Reply:
@ioannisg currently, I have moved around these 4 sections among text/rodata segment because of MMU requirement, namely, `.exc_vector_table`, `IRQ_VECTOR_TABLE`, `.openocd_dbg` and `<linker_relocate.ld>`.  There 4 are sections read only. However, from MMU point of view, these read only sections are either executable or never-execute. We have to classify accordingly.

Some background on MMU implementation:
MMU will mark "text" segment `[_image_text_start, _image_text_end]` as code, i.e. read only and executable.
And it will mark "rodata" segment `[_image_rodata_start, _image_rodata_end]` as rodata, i.e. read only and never execute.

Justification for change:
When you suggest to make aarch64 linker file same as cortex-M or cortex-R, you are suggesting to put vector table in "read only" segment. From MMU point of view, "read only" segment is not executable, so we must put vector table in "text" segment. 

I can move other 3 sections to "rodata" segment- based on the assumption that all 3 are actually read only and never-execute. [currently, I have put only `IRQ_VECTOR_TABLE` in rodata, I can move remaining 2 also to rodata]

For cortex-R, MMU has not yet come into the picture. Once does, I think we may need similar changes there as well.

Reply:
No, i am not suggesting to put vectors in rodata; I suggested to follow the same structure as Cortex-R where .vectors are inside exc_vector_table section, and _that_ one is grouped inside rom_start.

But now I realize your point; MMU has more strict requirement regarding attributing segments, than Cortex-M; in Cortex-M we play around with the section grouping freely, because we do not define MPU sections in the flash (we only do it in SRAM).

In that case, yes, I'd rather see, then, everything that is just read-only (no exec) inside ro-data. You could even remove the sections if they are not needed (They could be added later if this becomes needed).

Anyway, this was just a point of organizing the linker script in a consistent way, I guess we can live with the current approach

Reply:
I think we are on the same page now. I will put `IRQ_VECTOR_TABLE` in rodata, `openocd_dbg` in rodata (I see some data constructs being put here by `subsys/debug/openocd.c`) and will skip adding `<linker_relocate.ld>` as `CONFIG_CODE_DATA_RELOCATION` is not defined for aarch64 currently(add later if needed).